mitodl/ocw-studio
# Updated app to use new site config structure

#### Pre-Flight checklist


#### What are the relevant tickets?
Closes #120 

#### What's this PR do?
- Adds a management command to override site configs in the local db
- Updates the front and back end to use the new config format (see issue for details), particularly in the site sidebar where content editing links are generated

#### How should this be manually tested?
- Use the new management command to update the local site config for the course site starter
- Modify the override config to add/remove new menu items, then make sure the sidebar looks correct and links to the correct pages with the correct forms

#### Any background context you want to provide?
- I added the YAML file and management command for overriding site configs in the db because the practice of making changes to the course site config, then creating data migrations to adapt the config JSON to match it was not sustainable. We should drop that practice and apply course site config changes in that override YAML file.
- I removed the page/resource constants because this new config is designed to be more generalized/arbitrary. We no longer have a need to limit WebsiteContent records to be either a page or a resource.

#### Screenshots (if appropriate)
![ss 2021-03-18 at 15 03 57 ](https://user-images.githubusercontent.com/14932219/111682832-33c4ab00-87fb-11eb-87b9-3dbb2ea0e2c9.png)
